projects/ngneat/reactive-forms/src/lib/operators/diff.ts

```diff
@@ -0,0 +1,49 @@
+import { MonoTypeOperatorFunction, Observable } from 'rxjs';
+import { filter, pairwise, startWith, map, tap } from 'rxjs/operators';
+import { compareArraysContent } from '../utils';
+
+const toArray = object => Object.keys(object);
+const isArray = (value): boolean => value && Array.isArray(value);
+const isObject = (value): boolean => typeof value === 'object' && value !== null;
+const isFormArray = (prev, curr): boolean => isArray(curr) || isArray(prev);
+const isFormGroup = (prev, curr): boolean => isObject(curr) || isObject(prev);
+const isFormControl = (prev, curr): boolean => !isFormArray(prev, curr) && !isFormGroup(prev, curr);
+
+/**
+ * An operator which is used to filter valueChanges$ output, that it would emit only changed parts.
+ * @return {MonoTypeOperatorFunction} An Observable that emits items from the source Observable with only changed values.
+ */
+export function diff<T>(): MonoTypeOperatorFunction<T> {
+  return (source$: Observable<T>) =>
+    source$.pipe(
+      startWith(undefined),
+      pairwise(),
+      map(control => reduceControlValue<T>(...control)),
+      filter(control => control !== undefined)
+    );
+}
+
+function reduceControlValue<T>(prev: T, curr: T): T {
+  if (prev === undefined) {
+    return curr;
+  }
+
+  if (isFormControl(prev, curr)) {
+    return prev === curr ? undefined : curr;
+  }
+
+  if (isFormArray(prev, curr)) {
+    return compareArraysContent(prev, curr) ? undefined : curr;
+  }
+
+  return reduceFormGroup(prev, curr);
+}
+
+function reduceFormGroup<T>(prev: T, curr: T): T {
+  const reduced = toArray(curr).reduce((acc, key) => {
+    const control = reduceControlValue(prev[key], curr[key]);
+    return control !== undefined ? { ...acc, [key]: control } : acc;
```

Comment:
There is no need to create a new object on each iteration.

Reply:
I nee help with this one, thanks.

Reply:
I mean:
```ts
control !== undefined ? acc[key] = control;

return acc;
```